### PR TITLE
feat: simplify `done()` function

### LINKED ISSUE #596: Simplify `done()` Function to Only Accept Boolean for Authentication Status
### Context
The current implementation of the `done()` function allows specifying custom error codes and messages, which might lead to inconsistencies and over complication, especially in standard HTTP authentication cases. This proposal is to simplify the `serverAuth` function to accept only a Boolean value indicating authentication success (`true`) or failure (`false`).

### Current Behavior
The `done()` function currently accepts the following parameters:
- A Boolean value indicating the authentication status.
- An optional HTTP status code.
- An optional error message.

Example:
```javascript
serverAuth({ authProps, done }) {
  // Various invocation possibilities
  // done(true)
  // done(false, 401, "Unauthorized")
  // done(false)
}
```

### Proposed Change
Simplify the `done()` function to accept only a Boolean value:
- `true` for successful authentication.
- `false` for failed authentication.

In cases of authentication failure, the function should automatically set the HTTP status code to `401 Unauthorized`.

### PR DESCRIPTION
**Pull request by ankur0904**
**Description**

- This PR simplified the `done()` function to take only one input `bool` as input.
- The related documentation is also changed.

**Related issue(s)**
Resolves #596

### CODE REVIEW

File: src/adapters/http/server.ts [hunk 1]
```diff
@@ -50,11 +50,11 @@ class HttpAdapter extends Adapter {
       })
       return {
         promise,
-        done: (val: boolean, code = 401, message = 'Unauthorized') => {
```
Comment (KhudaDad414) [inline src/adapters/http/server.ts:53]: can you do the same for the WebSocket adapter?

Comment (ankur0904) [reply src/adapters/http/server.ts:53]: @KhudaDad414 Sure :)

**Comment by ankur0904** [comment]
@KhudaDad414 The current code don't have `code = 401, message = 'Unauthorized'` these arguments where the changes is supposed to done.

Please tell me if I am missing something.

```JS
class HttpAdapter extends Adapter {
    ...
     // code of done function
      return {
        promise,
        done: (val: boolean) => {
          if (val) {
            resolveFunc(true)
          } else {
            rejectFunc({ code: 401, message: 'Unauthorized' })
          }
      }
   }
   ...
}
```

**Comment by KhudaDad414** [comment]
@ankur0904 ws adapter has the same code:

https://github.com/asyncapi/glee/blob/60484deec8da810e1177b6c585065d22627be87f/src/adapters/ws/server.ts#L221

**Comment by ankur0904** [comment]
Sorry, @KhudaDad414 I missed the `ws` folder. I made the commit in which the required changes are reflected. Please have a look.

**Comment by KhudaDad414** [comment]
@afzal442 this change will touch the docs. do you want to have look?

**Comment by afzal442** [comment]
sure! I do. and thanks for reminder.

**Comment by afzal442** [comment]
So now `done()` func handles boolean value. LGTM!

**Review by KhudaDad414 — APPROVED** [review]
LGTM!

**Comment by afzal442** [comment]
Thanks @ankur0904 for your contrib. 💪🏻

**Comment by ankur0904** [comment]
Thanks @KhudaDad414 @afzal442 for merging the PR.

**Pull request merged**